snakebite/client.py

```diff
@@ -1425,10 +1476,16 @@ def __handle_socket_error(self, exception):
         log.debug("Request failed with %s" % exception)
         if exception.errno in (errno.ECONNREFUSED, errno.EHOSTUNREACH):
             # if NN is down or machine is not available, pass it:
-            pass
+            if self.__should_retry():
+                return
+            else:
+                pass
         elif isinstance(exception, socket.timeout):
             # if there's communication/socket timeout, pass it:
-            pass
+            if self.__should_retry():
+                return
+            else:
+                pass
         else:
             raise
         self.namenode.next()
```

Comment:
Are there any other [socket.error](https://docs.python.org/2/library/socket.html#socket.error) except `timeout`, `errno.ECONNREFUSED` and `errno.EHOSTUNREACH` that we want to retry on?